### DISCUSSION TITLE
Send verification email with Auth API

### DISCUSSION THREAD

**Discussion by WilliamScott-AdventGX** [supabase/supabase#8062] [Questions]
Supabase-JS' Auth API lets you create accounts like so:
`client.auth.api.createUser({ email, password });`

However this code does not send a verification email. To send a verification email, I need to use the auth client code:
`client.auth.signUp({email, password});`

Am I missing something? **I would like the auth API to send a verification email on user creation.**
I am aware you can automate this yourself via your own email server, however, I would like to use the built-in email service.

Note: I do not want to disable email verification or skip verification; The user should verify their email.

**Comment by GaryAustin1** [comment]
``` 
 supabase.auth.api.createUser({
      email,
      password,
      email_confirm: true,
    });
```
https://supabase.com/docs/reference/javascript/auth-api-createuser
The language is a bit confusing in the docs, but I think you need the email_confirm:true or it does not send out confirmation.

**Reply by WilliamScott-AdventGX** [reply]
Was able to test it today. Typescript's typing for the variable read `Sets if a user has confirmed their email address.` Looks like it does not send an email.

**Reply by GaryAustin1** [reply]
Ah, that could be it.
Here are a few discussions, not sure they are much help:
https://github.com/supabase/gotrue/issues/413
https://github.com/supabase/gotrue/issues/492
https://github.com/supabase/gotrue/issues/438
